src/components/contracts/ExtendContractForm.vue

```diff
@@ -0,0 +1,360 @@
+  mdiFolderInformationOutline
+} from "@mdi/js";
+
+import { mapGetters } from "vuex";
+
+const worktimeNotZero = (value) => value !== "00:00";
+const validWorktime = (value) => {
+  try {
+    return value.split(":")[1] <= parseInt(59);
+  } catch {
+    return false;
+  }
+};
+const validCarryoverMinutes = (value) => {
+  return /^(-?)([0-9][0-9]):[0-5][0-9]$/.test(value);
+};
+
+export default {
+  name: "ExtendContractForm",
+  filters: {
+    formatDate(date) {
+      return localizedFormat(parseISO(date), "yyyy-MM-dd");
+    }
+  },
+  components: { ContractFormDateInput },
+  mixins: [contractExpiredMixin, validationMixin],
+  validations: {
+    contract: {
+      name: { required, maxLength: maxLength(100), minLength: minLength(2) },
+      carryoverMinutes: {
+        required,
+        validCarryoverMinutes,
+        minLength: minLength(5),
+        maxLength: maxLength(6)
+      },
+      worktime: {
+        required,
+        worktimeNotZero,
+        validWorktime,
+        minLength: minLength(5)
+      }
+    }
+  },
+  props: {
+    uuid: {
+      type: String,
+      default: null
+    },
+    entity: {
+      type: Object,
+      default: null
+    }
+  },
+  data: () => ({
+    icons: {
+      mdiFolderInformationOutline,
+      mdiTimetable,
+      mdiCalendar,
+      mdiCalendarClock
+    },
+    contract: null,
+    carryover: false,
+    carryoverDateMenu: false,
+    carryoverCache: null,
+    maxEndDate: null
+  }),
+
+  computed: {
+    ...mapGetters({
+      shifts: "shift/shifts"
+    }),
+    carryoverMinimalDate() {
+      return this.contract.date.start;
+    },
+    carryoverMaximalDate() {
+      return this.contract.date.end;
+    },
+    carryoverTargetDate() {
+      return format(parseISO(this.contract.carryoverTargetDate), "MMMM yyyy");
+    },
+    sortedShifts() {
+      const shifts = this.shifts.filter(
+        (shift) => shift.contract === this.entity.uuid
+      );
+      return shifts.sort((a, b) => {
+        return new Date(a.date.start) - new Date(b.date.start);
+      });
+    },
+    lastShiftOfContract() {
+      if (this.sortedShifts.length < 1) return this.startDate;
+
+      return format(
+        new Date(this.sortedShifts[this.sortedShifts.length - 1].date.start),
+        "yyyy-MM-dd"
+      );
+    },
+    firstShiftOfContract() {
+      if (this.sortedShifts.length < 1) return null;
+
+      return format(new Date(this.sortedShifts[0].date.start), "yyyy-MM-dd");
+    },
+    valid() {
+      if (
+        this.worktimeErrors.length > 0 ||
+        this.nameErrors.length > 0 ||
+        this.contract.name === null ||
+        this.contract.worktime === null ||
+        this.carryoverMinutesErrors.length > 0
+      ) {
+        return false;
+      }
+
+      return true;
+    },
+    initialData() {
+      const startDate = localizedFormat(startOfMonth(new Date()), "yyyy-MM-dd");
+
+      return new Contract({
+        name: null,
+        date: {
+          start: startDate,
```

Comment:
das müsste sich noch verändern - aktuell kommt hier das `startDate` des (alten) Quellvertrages heraus, es müsste aber quasi + 1 Tag sein, oder?

Reply:
Oh, ich glaube, ich habe das missverstanden... die "Verlängerungs"-Funktion legt keinen neuen Vertrag (= eine Kopie des alten mit neuen Daten) an, sondern sie soll nur das Enddatum in die Zukunft setzen.

Dann würde ich doch vorschlagen, keine neue Komponente zu bauen, sondern `ContractForm.Vue` zu nutzen und entsprechend zu erweitern:
* die entsprechenden `v-alerts` anzeigen
* alle Felder, die nicht geändert werden dürfen (start.date, worktime, name) deaktivieren
* Carryover (inkl. Checkbox) ganz ausblenden oder nur hier den Übertrag/letzten Stand des AZK zeigen

Reply:
Ich wollte eine extra Form erstellen, da ich nicht 80% der Felder Konditional machen wollte.